core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -805,11 +805,29 @@ protected function generateButtons($arrRow, $strTable, $arrRootIds=array(), $bln
 			{
 				if ($k == 'show')
 				{
-					$return .= '<a href="'.$this->addToUrl($v['href'].'&amp;id='.$arrRow['id'].'&amp;popup=1').'" title="'.\StringUtil::specialchars($title).'" onclick="Backend.openModalIframe({\'title\':\''.\StringUtil::specialchars(str_replace("'", "\\'", sprintf($GLOBALS['TL_LANG'][$strTable]['show'][1], $arrRow['id']))).'\',\'url\':this.href});return false"'.$attributes.'>'.\Image::getHtml($v['icon'], $label).'</a> ';
+					if ($v['route'])
+					{
+						$href = \System::getContainer()->get('router')->generate($v['route'], ['id' => $arrRow['id'], 'popup' => '1']);
+					}
+					else
+					{
+						$href = $this->addToUrl($v['href'].'&amp;id='.$arrRow['id'].'&amp;popup=1');
+					}
+
+					$return .= '<a href="'.$href.'" title="'.\StringUtil::specialchars($title).'" onclick="Backend.openModalIframe({\'title\':\''.\StringUtil::specialchars(str_replace("'", "\\'", sprintf($GLOBALS['TL_LANG'][$strTable]['show'][1], $arrRow['id']))).'\',\'url\':this.href});return false"'.$attributes.'>'.\Image::getHtml($v['icon'], $label).'</a> ';
 				}
 				else
 				{
-					$return .= '<a href="'.$this->addToUrl($v['href'].'&amp;id='.$arrRow['id'].(\Input::get('nb') ? '&amp;nc=1' : '')).'" title="'.\StringUtil::specialchars($title).'"'.$attributes.'>'.\Image::getHtml($v['icon'], $label).'</a> ';
+					if ($v['route'])
+					{
+						$href = \System::getContainer()->get('router')->generate($v['route'], ['id' => $arrRow['id']]);
```

Comment:
Did you omit the `Input::get('nb')` check here on purpose?

Reply:
Yeah, I assumed it is only relevant in a regular backend url. `nb` stands for "no buttons", which only works in the regular DC.

Reply:
But we can as well generate regular DC URLs with this feature, can't we? Would it hurt if we added it?

Reply:
What do you mean with "regular DC URLs"? It would always append the query parameter. As custom routes are Symfony controller to me, I don't think that parameter would ever make sense?